truenas/webui
# NAS-127892 / 24.10 / show validation errors for hidden controls

**Summary**

Adding `IxFormService`, which holds a collection of only those form fields, which are visible for a user.

That said, adding and removing control should is implemented using the `AfterViewInit` and `OnDestroy` methods of each `ix-` form component, such as `ix-checkbox`, `ix-input`, etc.

**Testing**

1. Go to **Shares > Windows SMB shares > Add SMB**
2. Enter **Name** and **Path** so that **Save** button become active
3. Click **Advanced** and check **Legacy AFP** checkbox, and click **Save.** The error will be displayed next to **Legacy AFP** checkbox, and **Save** button will become disabled. This behavior matches the expectations.
4. Click **Basic** , then go back to **Advanced**, and then click **Basic** again. This will make **Save** button clickable. This behavior also matches the expectations.
5. Now click the **Save** button. After a call to API, it will become become disabled, without displaying any error message.

   **Expected result:** error message is presented to the user, and it mentions to the field causing the **Save** button to be disabled
